### PR TITLE
TEST:Test coefficient coverage

### PR DESCRIPTION
**Pull request by ariba-sayed**
Added unit tests for the Coefficient explainer to improve test coverage.

Changes

1)Added tests for valid model initialization
2)Added tests for invalid model 
3) Verified correctness of attribution values
4)Verified output shape consistency
5) Covered edge case with empty input

The Coefficient explainer did not have dedicated tests.

All code and test logic have been reviewed and understood before submission.

### CODE REVIEW

File: tests/explainers/other/test_coefficient.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    result = explainer.attributions(X)
+
+    expected = np.array([[1, 2, 3], [1, 2, 3]])
+
+    assert np.array_equal(result, expected)
+
+
+# Ensures output shape is correct
+def test_attributions_shape():
```
Comment (CloseChoice) [inline tests/explainers/other/test_coefficient.py:49]: pls get rid of this test, it does not provide any value above `test_attributions_basic`

File: tests/explainers/other/test_coefficient.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+
+class BadModel:
+    pass
+
+
+# Checks that invalid input is rejected
+def test_init_invalid_model():
+    model = BadModel()
+    try:
```
Comment (CloseChoice) [inline tests/explainers/other/test_coefficient.py:24]: can you pls do this with `pytest.raises`?

File: tests/explainers/other/test_coefficient.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+# Checks that the class accepts a valid model
+class DummyModel:
+    coef_ = np.array([1, 2, 3])
+
+
+def test_init_valid_model():
+    model = DummyModel()
+    explainer = Coefficient(model)
+    assert explainer.model is model
```
Comment (CloseChoice) [inline tests/explainers/other/test_coefficient.py:14]: can you pls check this in one of the tests below and get rid of this test?

**Review by CloseChoice — CHANGES_REQUESTED** [review]

File: tests/explainers/other/test_coefficient.py (hunk 3 above)
Comment (24f3004086) [reply tests/explainers/other/test_coefficient.py:14]: yes i will do it

File: tests/explainers/other/test_coefficient.py (hunk 2 above)
Comment (24f3004086) [reply tests/explainers/other/test_coefficient.py:24]: yes
